player/common/src/main/kotlin/com/tidal/sdk/player/common/model/BaseMediaProduct.kt

```diff
@@ -6,4 +6,5 @@ interface BaseMediaProduct {
     val productId: String
     val sourceType: String?
     val sourceId: String?
+    val extras: Map<String, String?>
```

Comment:
Why restrict the values to the String type (and optional?)?

In iOS I was thinking of going with this
```
let extras: [String: Codable]
```

Main diference is that it allows any value that can be JSON encoded/decoded, and also make it non optional.
Any reason why you think we would want to have a key with a `null` value?

Reply:
Player uses an internal serialization stack which doesn't share configuration with that of the host. With your approach, if a host passes something that this stack knows how to serialize, there is no problem. But if they pass for example an enum value which requires a custom serializer, we won't know what to do with it. Requiring String solves this problem (even if it means you can't send non-String values, but that's easy to work around on the receiving end).

As for null values, we don't have constraints on values so I can't assume that no deserializer on the backend would consider the lack of a key equal to its value being null, which is why I allowed null keys.

Reply:
Thanks for the explanation.
On iOS we don't have such a limitation with serialization but I agree with using `String`
About nullability, that would be a weird backend implementation haha, but I have seen it before, so I also buy your explanation.